pynpoint/processing/psfpreparation.py

```diff
@@ -123,32 +131,40 @@ def run(self) -> None:
 
         # Run the PSFpreparationModule for each subset of frames
         start_time = time.time()
-        for i, _ in enumerate(frames[:-1]):
+        for j in range(wavelength_dim):
+            for i, _ in enumerate(frames[:-1]):
 
-            # Print progress to command line
-            progress(i, len(frames[:-1]), 'Preparing images for PSF subtraction...', start_time)
+                # Print progress to command line
+                progress(i + j * len(frames[:-1]), wavelength_dim * len(frames[:-1]),
+                         'Preparing images for PSF subtraction...', start_time)
 
-            # Get the images and ensure they have the correct 3D shape with the following
-            # three dimensions: (batch_size, height, width)
-            images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                # Get the images and ensure they have the correct 3D shape with the following
+                # three dimensions: (batch_size, height, width)
+                if len(im_shape) < 4:
+                    images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                else:
+                    images = self.m_image_in_port[j, frames[i]:frames[i+1], ]
 
-            if images.ndim == 2:
-                warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
-                              'An extra dimension has been added.')
+                if images.ndim == 2:
+                    warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
+                                  'An extra dimension has been added.')
 
-                images = images[np.newaxis, ...]
+                    images = images[np.newaxis, ...]
 
-            # Apply the mask, i.e., set all pixels to 0 where the mask is False
-            images[:, ~mask] = 0.
+                # Apply the mask, i.e., set all pixels to 0 where the mask is False
+                images[:, ~mask] = 0.
 
-            # If desired, normalize the images using the Frobenius norm
-            if self.m_norm:
-                im_norm = np.linalg.norm(images, ord='fro', axis=(1, 2))
-                images /= im_norm[:, np.newaxis, np.newaxis]
-                norms.append(im_norm)
+                # If desired, normalize the images using the Frobenius norm
+                if self.m_norm:
+                    im_norm = np.linalg.norm(images, ord='fro', axis=(1, 2))
+                    images /= im_norm[:, np.newaxis, np.newaxis]
+                    norms.append(im_norm)
 
-            # Write processed images to output port
-            self.m_image_out_port.append(images, data_dim=3)
+                # Write processed images to output port
+                if len(im_shape) < 4:
+                    self.m_image_out_port.append(images, data_dim=3)
+                else:
+                    self.m_image_out_port[j, frames[i]:frames[i+1], ] = images
```

Comment:
If possible, then better to use `append` with `data_dim=4`, but please check if it is appended to the right axis. We can also update `core.dataio` if needed.

Reply:
This problem is a part of the one from line 115. I'll solve this together with the other one and considre this instence as resolved.